Kernel/Syscalls/mmap.cpp

```diff
@@ -590,19 +590,36 @@ ErrorOr<FlatPtr> Process::sys$msyscall(Userspace<void*> address)
     return 0;
 }
 
-ErrorOr<FlatPtr> Process::sys$msync(Userspace<void*> address, size_t size, [[maybe_unused]] int flags)
+ErrorOr<FlatPtr> Process::sys$msync(Userspace<void*> address, size_t size, int flags)
 {
+    // Either MS_SYNC or MS_ASYNC shall be specified
+    // Only MS_SYNC, MS_ASYNC and MS_INVALIDATE are valid bits to be set in flag
+    if (((flags & MS_ASYNC) == MS_ASYNC) != ((flags & MS_SYNC) == MS_SYNC)
+        || (flags & (MS_SYNC | MS_ASYNC | MS_INVALIDATE)) != flags)
+        return EINVAL;
+    // Address shall be page aligned
+    if (address.ptr() % PAGE_SIZE != 0)
+        return EINVAL;
+    // Note: This is not specified
+    size = Memory::page_round_up(size);
+
     // FIXME: We probably want to sync all mappings in the address+size range.
-    auto* region = address_space().find_region_from_range(Memory::VirtualRange { address.vaddr(), size });
+    auto* region = address_space().find_region_containing(Memory::VirtualRange { address.vaddr(), size });
+    // All regions from address upto address+size shall be mapped
```

Comment:
```suggestion
```

Reply:
Not finding \*a\* region can have multiple causes, especially, once we support syncing multiple regions at once,
so the code cannot fully represent the intent as of now